src/EditorFeatures/Core.Wpf/InlineDiagnostics/InlineDiagnosticsTaggerProvider.cs

```diff
@@ -0,0 +1,109 @@
+    internal class InlineDiagnosticsTaggerProvider : AbstractDiagnosticsAdornmentTaggerProvider<InlineDiagnosticsTag>
+    {
+        private readonly IEditorFormatMap _editorFormatMap;
+        protected sealed override IEnumerable<PerLanguageOption2<bool>> PerLanguageOptions => SpecializedCollections.SingletonEnumerable(InlineDiagnosticsOptions.EnableInlineDiagnostics);
+
+        protected internal override bool IsEnabled => true;
+
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public InlineDiagnosticsTaggerProvider(
+            IThreadingContext threadingContext,
+            IEditorFormatMapService editorFormatMapService,
+            IDiagnosticService diagnosticService,
+            IAsynchronousOperationListenerProvider listenerProvider)
+            : base(threadingContext, diagnosticService, listenerProvider)
+        {
+            _editorFormatMap = editorFormatMapService.GetEditorFormatMap("text");
+        }
+
+        protected internal override bool IncludeDiagnostic(DiagnosticData diagnostic)
+        {
+            return
+                diagnostic.Severity is DiagnosticSeverity.Warning or DiagnosticSeverity.Error &&
+                !string.IsNullOrWhiteSpace(diagnostic.Message) && !diagnostic.IsSuppressed;
+        }
+
+        /// <summary>
+        /// Creates the InlineErrorTag with the error distinction
+        /// </summary>
+        protected override InlineDiagnosticsTag? CreateTag(Workspace workspace, DiagnosticData diagnostic)
+        {
+            Debug.Assert(!string.IsNullOrWhiteSpace(diagnostic.Message));
+            var errorType = GetErrorTypeFromDiagnostic(diagnostic);
+            if (errorType is null)
+            {
+                return null;
+            }
+
+            if (diagnostic.DocumentId is null)
+            {
+                return null;
+            }
+
+            var document = workspace.CurrentSolution.GetRequiredDocument(diagnostic.DocumentId);
+            if (document is null)
+            {
+                return null;
+            }
+
+            var locationOption = workspace.Options.GetOption(InlineDiagnosticsOptions.Location, document.Project.Language);
+            var navigateService = workspace.Services.GetRequiredService<INavigateToLinkService>();
+            return new InlineDiagnosticsTag(errorType, diagnostic, _editorFormatMap, locationOption, navigateService);
+        }
+
+        private static string? GetErrorTypeFromDiagnostic(DiagnosticData diagnostic)
+        {
+            if (diagnostic.Severity == DiagnosticSeverity.Error)
+            {
+                if (diagnostic.CustomTags.Contains(WellKnownDiagnosticTags.EditAndContinue))
+                {
+                    return EditAndContinueErrorTypeDefinition.Name;
+                }
+                else
+                {
+                    return PredefinedErrorTypeNames.SyntaxError;
+                }
```

Comment:
```suggestion
                return diagnostic.CustomTags.Contains(WellKnownDiagnosticTags.EditAndContinue)
                    ? EditAndContinueErrorTypeDefinition.Name
                    : PredefinedErrorTypeNames.SyntaxError;
```

Reply:
This tends to be easier to understand that you will always be returning something than having an if/else block.

Reply:
changed